inc/c_util/slist.h

```diff
@@ -29,31 +29,31 @@ typedef bool (*SLIST_MATCH_FUNCTION)(PSLIST_ENTRY list_entry, void* match_contex
+typedef bool (*SLIST_CONDITION_FUNCTION)(PSINGLYLINKEDLIST_ENTRY list_entry, const void* match_context, bool* continue_processing);
 
 /**
 * @brief                          Function passed to slist_for_each, which is called for the value of each node of the list.
 * @param list_entry               Value of the current list node being processed.
 * @param action_context           Context passed to slist_for_each.
 * @param continue_processing      Indicates if slist_for_each shall continue iterating through the next nodes of the list or stop.
 */
-typedef int (*SLIST_ACTION_FUNCTION)(PSLIST_ENTRY list_entry, void* action_context, bool* continue_processing);
+typedef int (*SLIST_ACTION_FUNCTION)(PSINGLYLINKEDLIST_ENTRY list_entry, const void* action_context, bool* continue_processing);
 
 #include "umock_c/umock_c_prod.h"
 
 #ifdef __cplusplus
 extern "C"
 {
 #endif
-MOCKABLE_FUNCTION(, bool, slist_initialize, PSLIST_ENTRY, list_head);
-MOCKABLE_FUNCTION(, bool, slist_is_empty, const PSLIST_ENTRY, list_head);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_add, PSLIST_ENTRY, list_head, PSLIST_ENTRY, list_entry);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_add_head, PSLIST_ENTRY, list_head, PSLIST_ENTRY, list_entry);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_remove, PSLIST_ENTRY, list_head, PSLIST_ENTRY, list_entry);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_remove_head, PSLIST_ENTRY, list_head);
-MOCKABLE_FUNCTION(, PSLIST_ENTRY, slist_find, PSLIST_ENTRY, list_head, SLIST_MATCH_FUNCTION, match_function, void*, match_context);
-MOCKABLE_FUNCTION(, int, slist_remove_if, PSLIST_ENTRY, list_head, SLIST_CONDITION_FUNCTION, condition_function, void*, match_context);
-MOCKABLE_FUNCTION(, int, slist_for_each, PSLIST_ENTRY, list_head, SLIST_ACTION_FUNCTION, action_function, void*, action_context);
+MOCKABLE_FUNCTION(, bool, slist_initialize, PSINGLYLINKEDLIST_ENTRY, list_head);
+MOCKABLE_FUNCTION(, bool, slist_is_empty, const PSINGLYLINKEDLIST_ENTRY, list_head);
+MOCKABLE_FUNCTION(, PSINGLYLINKEDLIST_ENTRY, slist_add, PSINGLYLINKEDLIST_ENTRY, list_head, PSINGLYLINKEDLIST_ENTRY, list_entry);
+MOCKABLE_FUNCTION(, PSINGLYLINKEDLIST_ENTRY, slist_add_head, PSINGLYLINKEDLIST_ENTRY, list_head, PSINGLYLINKEDLIST_ENTRY, list_entry);
+MOCKABLE_FUNCTION(, int, slist_remove, PSINGLYLINKEDLIST_ENTRY, list_head, PSINGLYLINKEDLIST_ENTRY, list_entry);
+MOCKABLE_FUNCTION(, PSINGLYLINKEDLIST_ENTRY, slist_remove_head, PSINGLYLINKEDLIST_ENTRY, list_head);
+MOCKABLE_FUNCTION(, PSINGLYLINKEDLIST_ENTRY, slist_find, PSINGLYLINKEDLIST_ENTRY, list_head, SLIST_MATCH_FUNCTION, match_function, const void*, match_context);
+MOCKABLE_FUNCTION(, int, slist_remove_if, PSINGLYLINKEDLIST_ENTRY, list_head, SLIST_CONDITION_FUNCTION, condition_function, const void*, match_context);
+MOCKABLE_FUNCTION(, int, slist_for_each, PSINGLYLINKEDLIST_ENTRY, list_head, SLIST_ACTION_FUNCTION, action_function, const void*, action_context);
 #ifdef __cplusplus
```

Comment:
Nitpick: let's add a newline before the #ifdef

Reply:
done